src/node_crypto.cc

```diff
@@ -3912,6 +3914,73 @@ void KeyObject::GetAsymmetricKeyType(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(key->GetAsymmetricKeyType());
 }
 
+static inline void SetParam(Environment* env, Local<Object> params,
+                            const char* key, Local<Value> value) {
+  bool ok = params->DefineOwnProperty(
+      env->context(),
+      String::NewFromUtf8(env->isolate(), key, NewStringType::kNormal)
+          .ToLocalChecked(),
+      value,
+      PropertyAttribute::ReadOnly).ToChecked();
+  CHECK(ok);
+}
+
+Local<Value> KeyObject::GetAsymmetricParams() const {
+  CHECK_NE(this->key_type_, kKeyTypeSecret);
+
+  Local<Object> params = Object::New(env()->isolate());
+  EVP_PKEY* pkey = this->asymmetric_key_.get();
+  RSA* rsa;
+  DSA* dsa;
+  EC_KEY* ec;
+  const EC_GROUP* ec_group;
+  int ec_curve_id;
+  const char* ec_curve_name;
+
+  switch (EVP_PKEY_id(pkey)) {
+  case EVP_PKEY_RSA_PSS:
+    // TODO(tniessen): Also retrieve additional PSS params as soon as OpenSSL
+    // provides a way to do that.
+  case EVP_PKEY_RSA:
+    // TODO(tniessen): This should really be EVP_PKEY_get0_RSA, but OpenSSL does
+    // not support that for RSA-PSS.
+    rsa = reinterpret_cast<RSA*>(EVP_PKEY_get0(pkey));
+    SetParam(env(), params, "modulusLength",
+             Integer::New(env()->isolate(), RSA_bits(rsa)));
+    // TODO(tniessen): Add publicExponent.
+    break;
+    return env()->crypto_rsa_pss_string();
```

Comment:
Dead code. Was this supposed to go on line 3943?

Reply:
> Was this supposed to go on line 3943?

No, seems like I messed up copying `GetAsymmetricKeyType` in vim.